plugins/arcaniax_block_exporter.js

```diff
@@ -0,0 +1,450 @@
+			}
+			return (c === 'x' ? r : (r & 0x3 | 0x8)).toString(16);
+		});
+	}
+	
+	function getBehaviorFormat(formData){
+		var alpha = "";
+		var offset_entity = "";
+		var size_entity = "";
+		var offset_pick = "";
+		var size_pick = "";
+		
+		alpha = formData.alpha;
+		
+		if (!formData.entity_collision){
+			offset_entity = "0, 0, 0";
+			size_entity = "0, 0, 0";
+		}
+		else{
+			offset_entity = getOffset();
+			size_entity = getSize();
+		}
+		offset_pick = getOffset();
+		size_pick = getSize();
+		
+		return getBehaviorRaw(formData, alpha, offset_entity, size_entity, offset_pick, size_pick);
+	}
+	
+	function getOffset(){
+		var x = Math.max(Math.min(collisionOffset[0], bbMaxOffset),-bbMaxOffset);
+		var y = Math.max(Math.min(collisionOffset[1], bbMaxSize), bbMinSize);
+		var z = Math.max(Math.min(collisionOffset[2], bbMaxOffset),-bbMaxOffset);
+		
+		x = x - Math.max(Math.min(collision[0],bbMaxSize-Math.abs(x*2)),bbMinSize)/2;
+		z = z - Math.max(Math.min(collision[2],bbMaxSize-Math.abs(z*2)),bbMinSize)/2;
+		
+		return x + ", " + y + ", " + z;
+	}
+	
+	function getSize(){
+		var offsetX = Math.max(Math.min(collisionOffset[0], bbMaxOffset),-bbMaxOffset);
+		var offsetY = Math.max(Math.min(collisionOffset[1], bbMaxSize), bbMinSize);
+		var offsetZ = Math.max(Math.min(collisionOffset[2], bbMaxOffset),-bbMaxOffset);
+		
+		var x = Math.max(Math.min(collision[0],bbMaxSize-Math.abs(offsetX*2)),bbMinSize);
+		var y = Math.max(Math.min(collision[1],bbMaxSize-offsetY),bbMinSize);
+		var z = Math.max(Math.min(collision[2],bbMaxSize-Math.abs(offsetZ*2)),bbMinSize);
+		
+		return x + ", " + y + ", " + z;
+	}
+	
+	function getBehaviorRaw(formData, alpha, offset_entity, size_entity, offset_pick, size_pick){
+		var json =
+`{
+  "format_version": "1.16.100",
+  "minecraft:block": {
+    "description": {
+      "identifier": "${formData.id}"
+    },
+    "components": {
+      "minecraft:destroy_time": 4.0,
+      "minecraft:friction": 0.6,
+	  "minecraft:block_light_emission": 0.0,
+      "minecraft:block_light_absorption": 0.0,
+	  "minecraft:entity_collision": {
+		"origin": [${offset_entity}],
+		"size": [${size_entity}]
+	  },
+	  "minecraft:pick_collision": {
+		"origin": [${offset_pick}],
+		"size": [${size_pick}]
+	  },
+	  "minecraft:geometry": "geometry.${formData.file_name}",
+      "minecraft:material_instances" : {
+        "*" : {
+          "texture": "${formData.file_name}",
+          "render_method": "${alpha}"
+        }
+      }
+    }
+  }
+}`;
+		return json;
+	}
+	
+	function onError(err)
+		{
+			if (err)
+			{
+				console.log('File Writing Error: ', err);
+			}
+		}
+	
+	
+	Plugin.register('arcaniax_block_exporter', {
+		title: 'Bedrock Block Exporter',
+		icon: 'icon-format_block',
+		author: 'Arcaniax',
+		description: 'Helps making new Bedrock blocks (requires experimental mode)',
+		version: '0.1.1',
+		variant: 'both',
+
+		onload() {
+
+			Blockbench.addMenuEntry('Bedrock Block Exporter', 'icon-format_block', function() {
```

Comment:
Please use Actions here. You can find an example in many newer plugins, e. g. "Plaster".
`addMenuEntry` is a deprecated method from the old days of Blockbench.

Reply:
Changed to the newer method.